applications/GeoMechanicsApplication/custom_strategies/schemes/generalized_newmark_T_scheme.hpp

```diff
@@ -0,0 +1,104 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//                   Richard Faasse
+//
+
+#pragma once
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "solving_strategies/schemes/scheme.h"
+
+// Application includes
+#include "geo_mechanics_application_variables.h"
+#include "geomechanics_time_integration_scheme.hpp"
+
+namespace Kratos {
+
+template <class TSparseSpace, class TDenseSpace>
+class GeneralizedNewmarkTScheme
+    : public GeoMechanicsTimeIntegrationScheme<TSparseSpace, TDenseSpace> {
+public:
+    using BaseType = Scheme<TSparseSpace, TDenseSpace>;
+    using DofsArrayType = typename BaseType::DofsArrayType;
+    using TSystemMatrixType = typename BaseType::TSystemMatrixType;
+    using TSystemVectorType = typename BaseType::TSystemVectorType;
+    using LocalSystemVectorType = typename BaseType::LocalSystemVectorType;
+    using LocalSystemMatrixType = typename BaseType::LocalSystemMatrixType;
+
+    KRATOS_CLASS_POINTER_DEFINITION(GeneralizedNewmarkTScheme);
+
+    explicit GeneralizedNewmarkTScheme(double theta)
+        : GeoMechanicsTimeIntegrationScheme<TSparseSpace, TDenseSpace>(), mTheta(theta)
+    {
+    }
+
+    ~GeneralizedNewmarkTScheme() override = default;
+
+    int Check(const ModelPart& rModelPart) const override
+    {
+        KRATOS_TRY
+
+        Scheme<TSparseSpace, TDenseSpace>::Check(rModelPart);
```

Comment:
I would simplify this to (using one of the previously defined aliases):
```suggestion
        BaseType::Check(rModelPart);
```

Reply:
I actually prefer to not use this alias so it is at one glance clear which base function is called